src/Accelerators/NNPA/Dialect/ZHigh/ZHighOps/OpHelper.cpp

```diff
@@ -209,5 +212,229 @@ bool oneIsOfNHWCLayout(Type t1, Type t2) {
   return true;
 }
 
+/// Check if ONNXReshapeOp is reshaping 2D to 4D by tiling each input dimension.
```

Comment:
Code is kind of fine as is, it feels like one could use a single version that goes from small to big, and one big to small.

Reply:
It also feel that nothing here is Z-related. If so, it should find a place outside of NNPA subduers